supadata/client.py

```diff
@@ -54,6 +55,7 @@ def _request(self, method: str, path: str, **kwargs: Dict[str, Any]) -> Dict[str
             dict: Parsed JSON response
 
         Raises:
+            SupadataError: If the API request fails with a gateway error
             requests.exceptions.RequestException: If the API request fails
         """
```

Comment:
Is this accurate? The goal is to raise `SupadataError` in all cases - both API and gateway.